src/main/resources/static/templates/PathTraversal/LEVEL_1/PathTraversal.js

```diff
@@ -1,39 +1,42 @@
-function addingEventListenerToLoadImageButton() {
-  document.getElementById("loadButton").addEventListener("click", function () {
-    let url = getUrlForVulnerabilityLevel();
-    doGetAjaxCall(
-      appendResponseCallback,
-      url + "?fileName=" + document.getElementById("fileName").value,
-      true
-    );
-  });
-}
-addingEventListenerToLoadImageButton();
-
 function appendResponseCallback(data) {
+  let infoContainer = document.getElementById("Information");
+  // Clear previous contents
+  infoContainer.textContent = '';
+
   if (data.isValid) {
-    let tableInformation = '<table id="InfoTable">';
+    let table = document.createElement('table');
+    table.id = "InfoTable";
     let content = JSON.parse(data.content);
+
     if (content.length > 0) {
+      let thead = document.createElement('thead');
+      let headerRow = document.createElement('tr');
       for (let key in content[0]) {
-        tableInformation =
-          tableInformation + '<th id="InfoColumn">' + key + "</th>";
+        let th = document.createElement('th');
+        th.id = "InfoColumn";
+        th.textContent = key;
+        headerRow.appendChild(th);
       }
+      thead.appendChild(headerRow);
+      table.appendChild(thead);
     }
+
+    let tbody = document.createElement('tbody');
     for (let index in content) {
-      tableInformation = tableInformation + '<tr id="Info">';
+      let tr = document.createElement('tr');
+      tr.id = "Info";
       for (let key in content[index]) {
-        tableInformation =
-          tableInformation +
-          '<td id="InfoColumn">' +
-          content[index][key] +
-          "</td>";
+        let td = document.createElement('td');
+        td.id = "InfoColumn";
+        // Use textContent to prevent XSS
+        td.textContent = content[index][key];
+        tr.appendChild(td);
       }
-      tableInformation = tableInformation + "</tr>";
+      tbody.appendChild(tr);
     }
-    tableInformation = tableInformation + "</table>";
-    document.getElementById("Information").innerHTML = tableInformation;
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities